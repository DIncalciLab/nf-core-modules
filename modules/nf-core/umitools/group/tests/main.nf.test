nextflow_process {

    name "Test Process UMITOOLS_GROUP"
    script "../main.nf"
    process "UMITOOLS_GROUP"

    tag "modules"
    tag "modules_nfcore"
    tag "umitools"
    tag "umitools/group"

    setup {
        run("UMITOOLS_EXTRACT") {
            script "../../extract/main.nf"
            config "./nextflow.config"
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                        file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)
                    ]
                ]
                """
            }
        }

        run("BWA_INDEX") {
            script "../../../bwa/index/main.nf"
            process {
                """
                input[0] = [
                 [ id:'sarscov2'],
                 file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
             ]
                """
            }
        }

        run("BWA_MEM") {
            script "../../../bwa/mem/main.nf"
            process {
                """
                input[0] = UMITOOLS_EXTRACT.out.reads
                input[1] = BWA_INDEX.out.index
                input[2] = [ [], [] ]
                input[3] = true
                """
            }
        }

        run("SAMTOOLS_INDEX") {
            script "../../../samtools/index/main.nf"
            process {
                """
                input[0] = BWA_MEM.out.bam
                """
            }
        }

    }

    test("no_umi") {

        when {
            process {
                """
                input[0] = [[ id:'test'], // meta map
                    file(params.test_data_base + '/data/genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.test_data_base + '/data/genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                ]
                input[1] = true
                input[2] = true
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.bam,
                    process.out.tsv,
                    file(process.out.log[0][1]).readLines()[0..1],
                    process.out.versions
                    ).match() }
            )
        }

    }

    test("with_umi - bam") {

        when {
            process {
                """
                input[0] = BWA_MEM.out.bam.join(SAMTOOLS_INDEX.out.bai, by: [0])
                input[1] = true
                input[2] = false
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.bam,
                    file(process.out.log[0][1]).readLines()[0..1],
                    process.out.versions
                    ).match() }
            )
        }

    }

    test("with_umi - tsv") {

        when {
            process {
                """
                input[0] = BWA_MEM.out.bam.join(SAMTOOLS_INDEX.out.bai, by: [0])
                input[1] = true
                input[2] = true
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.tsv,
                    file(process.out.log[0][1]).readLines()[0..1],
                    process.out.versions
                    ).match() }
            )
        }

    }

    test("with_umi - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = BWA_MEM.out.bam.join(SAMTOOLS_INDEX.out.bai, by: [0])
                input[1] = true
                input[2] = true
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
