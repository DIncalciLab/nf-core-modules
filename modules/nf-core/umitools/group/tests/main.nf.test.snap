{
    "with_umi - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_grouped.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_grouped.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_grouped.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,1f4cec69f12f1759748ed167701b321c"
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_grouped.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_grouped.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_grouped.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1f4cec69f12f1759748ed167701b321c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-16T19:52:11.292296043"
    },
    "with_umi - tsv": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test_grouped.tsv:md5,d652eb6570057e9e709e8cac5f43d00c"
                ]
            ],
            [
                "# UMI-tools version: 1.1.5",
                "# output generated by group -I test.bam --output-bam -S test_grouped.bam -L test_grouped.log --group-out test_grouped.tsv --paired --random-seed=100"
            ],
            [
                "versions.yml:md5,1f4cec69f12f1759748ed167701b321c"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-16T20:01:27.19796961"
    },
    "with_umi": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_grouped.log:md5,1758ea02d8ed0fcc81dd379456f180b2"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_grouped.bam:md5,f576b264fd3e8457f5e3716b09fcca58"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_grouped.tsv:md5,d652eb6570057e9e709e8cac5f43d00c"
                    ]
                ],
                "3": [
                    "versions.yml:md5,1f4cec69f12f1759748ed167701b321c"
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_grouped.bam:md5,f576b264fd3e8457f5e3716b09fcca58"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_grouped.log:md5,1758ea02d8ed0fcc81dd379456f180b2"
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_grouped.tsv:md5,d652eb6570057e9e709e8cac5f43d00c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1f4cec69f12f1759748ed167701b321c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-16T19:57:23.817618505"
    },
    "with_umi - bam": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test_grouped.bam:md5,f576b264fd3e8457f5e3716b09fcca58"
                ]
            ],
            [
                "# UMI-tools version: 1.1.5",
                "# output generated by group -I test.bam --output-bam -S test_grouped.bam -L test_grouped.log --paired --random-seed=100"
            ],
            [
                "versions.yml:md5,1f4cec69f12f1759748ed167701b321c"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-16T20:01:05.763105045"
    },
    "no_umi": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test_grouped.bam:md5,fb90b49a90c2b3e8ddfedd9c95361625"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test_grouped.tsv:md5,1a3ccdc00df59fa89c79ad482980a003"
                ]
            ],
            [
                "# UMI-tools version: 1.1.5",
                "# output generated by group -I test.paired_end.sorted.bam --output-bam -S test_grouped.bam -L test_grouped.log --group-out test_grouped.tsv --paired --random-seed=100"
            ],
            [
                "versions.yml:md5,1f4cec69f12f1759748ed167701b321c"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-16T20:02:22.66987719"
    }
}